File: day-01/solve.go
package main
import (
	"os"
	"log"
)

func main () {
	file, err := os.Open("input")
	if err != nil {
		log.Fatal(err)
	}

	data := make([]byte, 7000)
	count, err := file.Read(data);
	if err != nil {
		log.Fatal(err)
	}

	santa_at := 0
	santa_first_visit_to_basement := -1
	for i :=0; i < count; i++ {
		if data[i] == '(' {
			santa_at += 1
		} else if data[i] == ')' {
			santa_at -= 1
		}
		if santa_at == -1 && santa_first_visit_to_basement == -1 {
			santa_first_visit_to_basement = i + 1
		}
	}

	log.Printf("part 1: %d", santa_at)
	log.Printf("part 2: %d", santa_first_visit_to_basement)
}

File: day-01/solve.pl
use v5.22;
local $/ = undef;
my @todo = split("", scalar(<>));
my %step = ( '(' => 1, ')' => -1 );
my $santa_at = 0;
my $santa_first_visit_to_basement;
for my $i (0..$#todo) {
    $santa_at+= $step{$todo[$i]};
    $santa_first_visit_to_basement //= ($i+1) if $santa_at == -1;
}
say "part 1: $santa_at";
say "part 2: $santa_first_visit_to_basement";




